near-sdk/src/types/gas.rs

```diff
@@ -0,0 +1,115 @@
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+
+/// Represents the amount of NEAR tokens in "gas units" which are used to fund transactions.
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    PartialOrd,
+    Ord,
+    Eq,
+    BorshSerialize,
+    BorshDeserialize,
+    Hash,
+    BorshSchema,
+)]
+pub struct Gas(u64);
```

Comment:
```suggestion
pub struct Gas(pub u64);
```

We won't be changing the internal repr here, and the type doesn't have any invariants, so we might as well just expose this fact.

Reply:
hmm thought it would have been better to avoid exposing the inner type, but yeah I guess we can assume it will never change